### ISSUE TITLE
Question about the observation in evaluation_PersonaChat.py

### ISSUE THREAD

**Issue by Zhouziyi828** [Chenrj233/LMEDR#12]
I had a problem with evaluation_PersonaChat.py. I want to construct an _obervation_ dictionary and feed it into the _observe_ function. Then call the _act_ function to generate a reply. I don't understand the **"labels, label_candidates and episode_done"** parameters. I've seen them in PartAI, but can't relate them to this task. Could you please explain them? Or could you please provide an example?

Thank you very much!

**Comment by Chenrj233** [comment]
The **labels** indicate the target responses that we hope the model generates. The ConvAI2 dataset also includes a task of selecting candidate responses, where the **label_candidates** represent the responses provided for the model to choose from in the current conversation context. In the ConvAI2 dataset, each conversation consists of multiple turns, and it is important to differentiate between different conversations. The **episode_done** is used to mark the end of a conversation. For a more detailed introduction, please refer to the following links https://www.parl.ai/docs/tutorial_basic.html#, https://www.parl.ai/docs/core/messages.html. I hope the reply can be helpful to you.

**Comment by Zhouziyi828** [comment]
That means the "labe" is set when evaluating for comparison with the generated reply?

**Comment by Chenrj233** [comment]
Yes, during the evaluation phase, **labels** can be utilized to compare against the generated replies. Furthermore, **labels** also serve as dialogue context for the same conversation.

**Comment by Zhouziyi828** [comment]
Thank you for your reply.
I have another question about the training: when debugging, I found that the loss of "mem" and "bow" in each epoch did not change during the training process, may I ask what caused this?

**Comment by Chenrj233** [comment]
In our experiment, both bow and mem losses didn't change much in the first few epochs. However, once the lm loss dropped to a certain point, the bow and mem losses started to decrease. We guess that during the early stages of training, the model pays more attention to learning the structure of the conversation, so the lm loss drops more significantly.

**Comment by Zhouziyi828** [comment]
Thank you very much for your answer！
I have one more small question! 
When loading the LMEDRModel to train PersonaChat, num_labels is set to 1, but cand is set to 5. Are the two not related? The number of categories for each cand should not be 2, is it positive or negative? Why is it 1?

**Comment by Chenrj233** [comment]
It's important to note that the response selection task here doesn't involve classifying every candidate responses. Instead, it's about selecting the most suitable response from multiple candidate responses as the answer. "cand=5" indicates that for a given conversation history, the model needs to choose the most appropriate option from five candidate responses, without individually determining whether these candidate responses are the ground truth. Therefore, in the case of "num_labels=1", the model can score each response and then make the final choice based on those scores.

**Comment by Zhouziyi828** [comment]
> It's important to note that the response selection task here doesn't involve classifying every candidate responses. Instead, it's about selecting the most suitable response from multiple candidate responses as the answer. "cand=5" indicates that for a given conversation history, the model needs to choose the most appropriate option from five candidate responses, without individually determining whether these candidate responses are the ground truth. Therefore, in the case of "num_labels=1", the model can score each response and then make the final choice based on those scores.

Thank you for your reply.
I understand the meaning of cand, but I still don't understand the meaning of num_labels. In the source code, when cls_label=None, the final shape of cls_logits is (bs*cand,num_labels). I can't understand what num_labels means.

**Comment by Chenrj233** [comment]
In fact, when cls_label=None, the model is in the inference stage. We want to obtain scores for each candidate response, so we use a regression head with num_labels=1. Additionally, the model needs to select the final response based on cls_logits, which represents the scores of bs*cand responses. For more details about num_labels, you can refer to the documentation of [BartConfig](https://huggingface.co/docs/transformers/v4.30.0/en/model_doc/bart#transformers.BartConfig) and [BartForSequenceClassification](https://huggingface.co/docs/transformers/v4.30.0/en/model_doc/bart#transformers.BartForSequenceClassification).

**Comment by Zhouziyi828** [comment]
Thanks for your patient reply!!!
